sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SchemaPruning.scala

```diff
@@ -28,6 +31,7 @@ object SchemaPruning {
   def pruneDataSchema(
       dataSchema: StructType,
       requestedRootFields: Seq[RootField]): StructType = {
+    val resolver = sqlConf.resolver
```

Comment:
Combine L25, i.e.`val resolver = SQLConf.get.resolver`?

Reply:
This will always returns same resolver, it cannot be changed dynmanically

Reply:
Okay will do that